generators/generator-base.js

```diff
@@ -1773,6 +1773,15 @@ module.exports = class extends PrivateBase {
         return name.match(/^\d/) ? 'App' : name;
     }
 
+    /**
+     * get the Vue application name.
+     * @param {string} baseName of application
+     */
+    getVueAppName(baseName = this.baseName) {
```

Comment:
Instead of `vue`, or `angular` or `react` specific name, can we use generic name like `getFrontendAppName`

Reply:
I couldn't agree more.
In my opinion making needles and apis technology independent should be a priority jhipster 7 task.

Reply:
If we're merging the existing `getXXXName()` methods into one `getFrontendAppName()`, please let me know which implementation implementation should I choose. They are not same at the moment.

Reply:
@vivekmore there is only getAngularAppName and getAngularXAppName for frontend. React and vue are using angular version.
- `getAngularAppName -> getFrontendAppName`
- `getAngularXAppName -> getWebAppName` ?? It's used only in manifest.webapp.ejs, or drop and calculate in `manifest.webapp.ejs`
- Update variables `angularAppName -> frontendAppName`
- Move `manifest.webapp.ejs` from client/templates/*/src/main/webapp to `client/templates/common/src/main/webapp`. Move the reference to files-common too.

Others `get*Name` are server related (microservice, uaa).

Reply:
Ok 👍 @mshima 
Let me make the changes and update the PR

Reply:
@mshima, I didn't understand 
> The fallback to 'App' is not necessary since baseName is required.

did you mean appending `App` suffix is not required
```
const name = _.camelCase(baseName) + (baseName.endsWith('App') ? '' : 'App');
```

or enforcing names to start with a non numerical value is not required?

```
return name.match(/^\d/) ? 'App' : name;
```

Reply:
My bad. Removed from the list. You should let the way it is now.